x-pack/plugins/security_solution/server/lib/detection_engine/signals/find_threshold_signals.ts

```diff
@@ -50,69 +50,98 @@ export const findThresholdSignals = async ({
+            size: 10000, // max 10k buckets
+          },
+        });
+        if (i === (thresholdFields.length ?? 0) - 1) {
+          if (threshold.cardinality?.length) {
+            set(acc, `${aggPath}['aggs']`, {
+              top_threshold_hits: topHitsAgg,
+              cardinality_count: {
+                cardinality: {
+                  field: threshold.cardinality[0].field,
+                },
               },
-            };
-            // TODO: support case where threshold fields are not supplied, but cardinality is?
-            if (!isEmpty(threshold.cardinality_field)) {
-              set(acc, `${aggPath}['aggs']`, {
-                top_threshold_hits: topHitsAgg,
-                cardinality_count: {
-                  cardinality: {
-                    field: threshold.cardinality_field,
+              cardinality_check: {
+                bucket_selector: {
+                  buckets_path: {
+                    cardinalityCount: 'cardinality_count',
                   },
+                  script: `params.cardinalityCount >= ${threshold.cardinality[0].value}`, // TODO: cardinality operator
                 },
-                cardinality_check: {
-                  bucket_selector: {
-                    buckets_path: {
-                      cardinalityCount: 'cardinality_count',
+              },
+            });
+          } else {
+            set(acc, `${aggPath}['aggs']`, {
+              top_threshold_hits: topHitsAgg,
+            });
+          }
+        }
+        return acc;
+      }, {})
+    : {
+        threshold_0: {
+          terms: {
+            script: {
+              source: '""',
+              lang: 'painless',
+            },
+            min_doc_count: threshold.value,
+          },
+          aggs: {
+            top_threshold_hits: topHitsAgg,
+            ...(threshold.cardinality?.length
+              ? {
+                  cardinality_count: {
+                    cardinality: {
+                      field: threshold.cardinality[0].field,
                     },
-                    script: `params.cardinalityCount >= ${threshold.cardinality_value}`, // TODO: cardinality operator
                   },
-                },
-              });
-            } else {
-              set(acc, `${aggPath}['aggs']`, {
-                top_threshold_hits: topHitsAgg,
-              });
-            }
-          }
-          return acc;
-        }, {})
-      : {};
+                  cardinality_check: {
+                    bucket_selector: {
+                      buckets_path: {
+                        cardinalityCount: 'cardinality_count',
+                      },
+                      script: `params.cardinalityCount >= ${threshold.cardinality[0].value}`, // TODO: cardinality operator
+                    },
+                  },
+                }
+              : {}),
+          },
+        },
```

Comment:
The [bulk create codepath](https://github.com/elastic/kibana/pull/92667/files#diff-8dca2da82db2ec8d439d21b2741ef0ee405dc5a28d410ddf704d960c1c80149fR65) for the corresponding case where `threshold.field` is empty isn't parsing the aggregations. Should that branch of the `if` be using the cardinality agg that's created here? And if it's in a separate logic branch do we need the no-op `threshold_0` here?

Reply:
@marshallmain whoops, good catch... pushed up a fix.